transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -208,6 +211,15 @@ internal class MessagingClientImpl(
             }
     }
 
+    @Throws(Exception::class)
+    override suspend fun logoutFromAuthenticatedSession() {
+        if (authJwt == null) {
+            log.w { "Logout from anonymous session is not supported." }
+            return
+        }
+        logoutUseCase.logout(authJwt!!.jwt)
```

Comment:
I see the not-null assertion is used because smart-cast isn't possible here due to `authJwt` mutability. Would it be possible to use a shadow here? e.g.:

```
    @Throws(Exception::class)
    override suspend fun logoutFromAuthenticatedSession() {
        val authJwt = authJwt ?: run {
            log.w { "Logout from anonymous session is not supported." }
            return
        }
        logoutUseCase.logout(authJwt.jwt)
    }
```

Reply:
will do